### DISCUSSION TITLE
standards as prerequisite for federation

### DISCUSSION THREAD

**Discussion by jlp2duke** [AgBioData/DataFederation_WG#9] [Ideas]
Databases who would like to be connected can partake in an standardization scheme e.g. ISO20000 level of API documentation, metadata completeness and etc. to make connecting and writing parsers easier. These standards could be community driven and AgBioData has the wide network to start such an initiative.

**Comment by BrapiCoordinatorSelby** [comment]
I'm not sure exactly what ISO 20000 is or why it has any relevance here, but I agree with the sentiment. 

Data sharing requires at least two groups to agree on some standard for data structure (data types, required/optional information, standard vocabularies, transfer protocols, etc). Passing any information from one system to another always requires a negotiation, who is willing to put in the time and effort to change their own system to accommodate the flow of information. Any kind of data standard makes the negotiation easier by giving all parties a starting place to build from. 

True Data Federation is an extension of this problem, with many different systems trying to communicate to the outside world seamlessly as a unit. Whether or not it is planned, a data standard will develop naturally as each new group to join the federation will be pushed to follow the patterns established by the existing members. Then the question simply becomes: is the data standard well documented? A well documented, generalized standard will make it easier for more groups to join a federation, vs a data standard built from tribal knowledge and oral history will make it much harder for new systems to join the federation.

**Reply by BrapiCoordinatorSelby** [reply]
Use OpenAPI3 as standard for good, well structured, complete API documentation (https://swagger.io/specification/)
